### PR TITLE
feat: sm-21 - Create avatar component

### PR DESCRIPTION
**Pull request by svipulc**
#### Overview
This pull request introduces a fully functional `Avatar` component that supports displaying user images or initials. The component includes loading states, size variations, and error handling for invalid inputs. It leverages the `class-variance-authority` library for styling and supports dynamic class generation based on props.

#### Changes Made
1. **Avatar Component**:
   - Created a main `Avatar` component that accepts props for `children`, `isLoading`, `size`, and `src`.
   - Implemented error handling to ensure that either `src` or `children` is provided, but not both. If both are provided, an error is logged to the console.
   - Utilized conditional rendering to display:
     - A loading state with a fallback component.
     - An image if a valid URL is provided.
     - Initials derived from a name string if the `src` is not a valid URL.
     - Children if no `src` is provided.

2. **Avatar.Image Subcomponent**:
   - Created a subcomponent to handle rendering the avatar image with appropriate styling.
   - Utilized the same size variants as the main `Avatar` component.

3. **Avatar.Fallback Subcomponent**:
   - Created a fallback subcomponent that displays a loading animation or placeholder when the avatar image is not available.

4. **Utility Functions**:
   - **validateImage**: A utility function to validate if the provided `src` is a valid URL.
   - **getInitials**: A utility function to extract initials from a full name, returning the first letter of the first name and the first letter of the last name.

5. **Styling**:
   - Used `class-variance-authority` to manage styles for the avatar and its subcomponents, providing flexibility for size and variant management.

### CODE REVIEW

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -30,18 +30,18 @@ const buttonStyles = cva(
...
         colorscheme: "primary",
         className:
-          "text-primary-600 border-primary-500 bg-transparent hover:bg-primary-100",
+          "text-primary border-primary bg-transparent hover:bg-primary",
       },
       {
         variant: "ghost",
         colorscheme: "primary",
-        className: "text-primary-600 bg-transparent hover:bg-primary-100",
+        className: "text-primary bg-transparent hover:bg-primary",
```
Comment (adarshpandya-simform) [inline src/components/Button/index.tsx:44]: From the PR description, it seems like this PR intends to add `Avatar` component. Do you mind explaining the rationale behind changing the styles of the `Button` component here?

File: src/components/Avatar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
+// Avatar Component
+
+import { cn } from "@/utils";
+import { cva, VariantProps } from "class-variance-authority";
+import React, { ComponentProps } from "react";
+
+// Avatar
+const AvatarStyles = cva(
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:8]: Let's move styling to last or create a separate file. From a personal experience, most of the time, developers will open this file to see the component structure. Thoughts?

Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:8]: nit: 👇 

```suggestion
const avatarStyles = cva(
```

File: src/components/Avatar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+type AvatarProps = CustomAvatarProps & VariantProps<typeof AvatarStyles>;
+
+export const Avatar = ({
+  children,
+  isLoading,
+  size,
+  src,
+  ...props
+}: AvatarProps) => {
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:51]: Let's use `React.FC` here 👇 

```suggestion
export const Avatar: React.FC<AvatarProps> = ({
  children,
  isLoading,
  size,
  src,
  ...props
}) => {
```

File: src/components/Avatar/index.tsx [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+      {isLoading ? (
+        <Avatar.fallback size={size} />
+      ) : src && checkUrlOrNot(src) ? (
+        <Avatar.Image src={src} size={size} {...props} />
+      ) : src && !checkUrlOrNot(src) ? (
+        <>{src && getInitials(src)}</>
+      ) : (
+        children
+      )}
+    </div>
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:74]: Let's not use nested ternaries, I suggest a refactor here 😄

File: src/components/Avatar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+  );
+};
+
+// Avatar.Image
+const AvatarImageStyle = cva(
+  [
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:80]: Minor nit: 👇 
```suggestion
const avatarImageStyles = cva(
  [
```

File: src/components/Avatar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+  VariantProps<typeof AvatarImageStyle>;
+
+Avatar.Image = ({ src, size, ...props }: AvatarImageProps) => {
+  return (
+    <img src={src} className={cn(AvatarImageStyle({ size }))} {...props} />
+  );
+};
+
+// Avatar.Fallback
+const AvatarFallbackStyle = cva(
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:123]: Nit 👇 

```suggestion
const avatarFallbackStyles = cva(
```

File: src/components/Avatar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+Avatar.fallback = ({ size, ...props }: AvatarFallbackProps) => {
+  return <div className={cn(AvatarFallbackStyle({ size }))} {...props}></div>;
+};
+
+// Function use in Avatar component
+
+// Check if the input string is a valid URL
+function checkUrlOrNot(src: string) {
+  return /^(http|https):\/\/[^ "]+$/.test(src);
+}
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:169]: @svipulc Would it be a good idea to use native [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) constructor here?

File: src/components/Avatar/index.tsx
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:1]: Should we expose this component from [`index.tsx`](https://github.com/svipulc/Evoke-ui/blob/main/src/components/index.tsx)?

**Review by adarshpandya-simform — COMMENTED** [review]
I'd advice attaching snapshot for visual reference and please add an entry in CHANGELOG file for this component. Some refactor is needed.

File: src/components/Button/index.tsx (hunk 1 above)
Comment (svipulc) [reply src/components/Button/index.tsx:44]: The reason behind it that we have define our own theme in tailwind config file according to that we don't have "primary-600" in theme so if don't change storybook will not show css on avatar component. This button is not defined by me it come with storybook default boilerplate.

File: src/components/Avatar/index.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  alt,
+  fallback,
+  className,
+  ...props
+}) => {
+  return (
+    <div className={cn(avatarFallbackStyle({ size }), className)} {...props}>
+      {fallback && fallback}
+      {alt && !fallback && getInitials(alt)}
+      {!alt && !fallback && "?"}
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:93]: @svipulc Why do we need `"?"` here? It'd be truthy value always AFAIK?

File: src/components/Avatar/index.tsx [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+export const AvatarImage: React.FC<AvatarImageProps> = ({
+  src,
+  size = "md",
+  alt = "Alternative Text",
+  fallback,
+  className,
+  ...props
+}) => {
+  if (!src || src === "") {
+    return <AvatarFallback size={size} alt={alt} fallback={fallback} />;
```
Comment (adarshpandya-simform) [inline src/components/Avatar/index.tsx:58]: @svipulc can this be sufficient?

```suggestion
  if (!src) {
    return <AvatarFallback size={size} alt={alt} fallback={fallback} />;
```

**Review by adarshpandya-simform — COMMENTED** [review]
@svipulc Out of curiosity, why we shifted from compound component pattern that was implemented previously? 🤔

**Comment by adarshpandya-simform** [comment]
@svipulc Is it a visual bug? IMHO, Avatars should maintain their shape while reducing or increasing the size 👇 

https://github.com/user-attachments/assets/8946cd48-0145-4ba8-a0af-d01a511ccd0f

**Comment by adarshpandya-simform** [comment]
@svipulc Not sure about this but it seems like the size is not working for some stories 👇 


https://github.com/user-attachments/assets/9cdd2321-7d0b-4cc9-a699-5c9b1488256b

Is it a component-level bug? Or maybe some configuration we have missed in storybook?

**Pull request merged**